Assets/ModScripts/Submodules/PianoDecryption.cs

```diff
@@ -0,0 +1,226 @@
+    }
+
+    public override void OnPianoPress(int Piano)
+    {
+        if (Module.IsAnimating())
+            return;
+
+        Module.Audio.PlaySoundAtTransform(Module.PianoSounds[Piano + (Info.Piano * 12)].name, Module.transform);
+        Module.Piano[Piano].GetComponentInChildren<KMSelectable>().AddInteractionPunch(0.25f);
+
+        if (Module.IsModuleSolved())
+            return;
+
+        if (!Module.CheckValidComponents())
+        {
+            Debug.LogFormat("[The Cruel Modkit #{0}] Strike! The {1} key on the piano was pressed when the component selection was [{2}] instead of [{3}].", ModuleID, Info.PianoKeyNames[Piano], Module.GetOnComponents(), Module.GetTargetComponents());
+            Module.CauseStrike();
+            return;
+        }
+
+        Module.StartSolve();
+        if (Piano != solutionSequence[repeats > 1 ? currentNote % 4 : currentNote])
+        {
+            Debug.LogFormat("[The Cruel Modkit #{0}] Strike! The {1} key was pressed instead of {2}. Resetting input.", ModuleID, Info.PianoKeyNames[Piano], Info.PianoKeyNames[solutionSequence[currentNote]]);
+            Module.CauseStrike();
+            currentNote = 0;
+            return;
+        }
+        currentNote++;
+        if (currentNote == solutionSequence.Length * repeats)
+        {
+            Debug.LogFormat("[The Cruel Modkit #{0}] The entire piano sequence was submitted. Module solved.", ModuleID);
+            Module.Solve();
+            return;
+        }
+    }
+
+    public override void OnUtilityPress()
+    {
+        if (Module.IsAnimating())
+            return;
+
+        Module.Audio.PlayGameSoundAtTransform(KMSoundOverride.SoundEffect.ButtonPress, Module.transform);
+        Module.UtilityButton.GetComponentInChildren<KMSelectable>().AddInteractionPunch(0.5f);
+
+        if (Module.IsModuleSolved())
+            return;
+
+        if (!Module.CheckValidComponents())
+        {
+            Debug.LogFormat("[The Cruel Modkit #{0}] Strike! The ❖ button was pressed when the component selection was [{1}] instead of [{2}].", ModuleID, Module.GetOnComponents(), Module.GetTargetComponents());
+            Module.CauseStrike();
+            return;
+        }
+
+        Module.StartSolve();
+        currentNote = 0;
+    }
+
+    void EncryptWord()
+    {
+        for (int i = 0; i < WordList.Length; i++)
+        {
+            List<string> Encryption = new List<string>() { };
+            encryptedWord = WordList[i];
+            decryptedWord = encryptedWord;
+            Encryption.Add(encryptedWord);
+
+            encryptedWord = AtbashCipher(encryptedWord);
+            Encryption.Add(encryptedWord);
+
+            string rot13 = Rot13Cipher(encryptedWord);
+            char[] serialLetters = Module.GetComponent<KMBombInfo>().GetSerialNumberLetters().ToArray();
+            if (encryptedWord.Any(x => serialLetters.Contains(x)) & !rot13.Any(x => serialLetters.Contains(x))) continue;
```

Comment:
Here, you're checking if your Rot13 string has SN letters. If there aren't, you change words. Why? You can just ignore the rot13 part.